cloud/linode/client/client.go

```diff
@@ -48,14 +57,22 @@ type Client interface {
 var _ Client = (*linodego.Client)(nil)
 
 // New creates a new linode client with a given token, userAgent, and API URL
-func New(token, userAgent, apiURL string) (*linodego.Client, error) {
-	linodeClient := linodego.NewClient(nil)
+func New(token, userAgent, apiURL string, timeout time.Duration) (*linodego.Client, error) {
+	tokenSource := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: token})
+	oauth2Client := &http.Client{
+		Transport: &oauth2.Transport{
+			Source: tokenSource,
+		},
+		Timeout: timeout,
+	}
+	linodeClient := linodego.NewClient(oauth2Client)
 	client, err := linodeClient.UseURL(apiURL)
 	if err != nil {
 		return nil, err
 	}
 	client.SetUserAgent(userAgent)
 	client.SetToken(token)
```

Comment:
I think we can remove this line if we're setting the token in the oauth2 client?

Reply:
Updated